src/Bicep.Decompiler/TemplateDecompiler.cs

```diff
@@ -32,15 +32,16 @@ public class TemplateDecompiler
         private readonly IFileResolver fileResolver;
         private readonly IModuleRegistryProvider registryProvider;
         private readonly IConfigurationManager configurationManager;
-        private readonly ApiVersionProvider apiVersionProvider = new ApiVersionProvider();
+        private readonly IApiVersionProvider apiVersionProvider;
 
-        public TemplateDecompiler(IFeatureProvider features, INamespaceProvider namespaceProvider, IFileResolver fileResolver, IModuleRegistryProvider registryProvider, IConfigurationManager configurationManager)
+        public TemplateDecompiler(IFeatureProvider features, INamespaceProvider namespaceProvider, IFileResolver fileResolver, IModuleRegistryProvider registryProvider)
         {
             this.features = features;
             this.namespaceProvider = namespaceProvider;
             this.fileResolver = fileResolver;
             this.registryProvider = registryProvider;
-            this.configurationManager = configurationManager;
+            this.configurationManager = IConfigurationManager.ForConfiguration(IConfigurationManager.GetBuiltInConfiguration().WithAllAnalyzersDisabled());
```

Comment:
Why not just pass in as a constructor arg?

Reply:
The diff is obscuring what's going on in this class: the existing decompiler only uses `configurationManager` to get a copy of the built-in configuration with all analyzers disabled, then passes that along to a `ModuleDispatcher` and `Compilation` instance. Since `GetBuiltInConfiguration` was made a static interface method in this PR, I'll remove this from the constructor so the diff is clearer.